docs/architecture/grpc-for-wcf-developers/call-credentials.md

```diff
@@ -1,11 +1,143 @@
 ---
 title: Call credentials - gRPC for WCF Developers
-description: TO BE WRITTEN
+description: How to implement and use gRPC call credentials in ASP.NET Core 3.0
 author: markrendle
 ms.date: 09/02/2019
 ---
 
 # Call credentials
 
+Call credentials are all based on some kind of token passed in metadata with each request.
+
+## WS-Federation
+
+ASP.NET Core supports WS-Federation using the [WsFederation](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.WsFederation) package. This is the closest thing to Windows Authentication that is supported over HTTP/2. Users are authenticated using Active Directory Federation Services (ADFS), which provides a token that can be used to authenticate with ASP.NET Core.
```

Comment:
This is the closest thing -> This what? This package is the closest thing?

Reply:
Yes. I don't want to write "... package. This package...".

Reply:
Changed to "... package. WS-Federation is the closest ..."